### PR TITLE
General geometry collection

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/geometry/collections/collection.py [hunk 1]
```diff
@@ -2,15 +2,240 @@
...
+    # data
+    # ==========================================================================
+
+    @property
+    def DATASCHEMA(self):
+        """schema.Schema : Schema of the data representation."""
+        from schema import Schema
+        return Schema({
+            'itype': '.../...',
+            'items': lambda item: 'validate schema?'
```
Comment (gonzalocasas) [inline src/compas/geometry/collections/collection.py:92]: This definitely needs an `itype` before merging, as well as something in the `items` stuff.

Comment (gonzalocasas) [reply src/compas/geometry/collections/collection.py:92]: Actually... what is `itype`? :P `item_type` I assume, but it's treated used like the `dtype` of the data serialization framework with the same 2-level contraint, which I'm not sure makes so much sense here.

**Review by gonzalocasas — COMMENTED** [review]
After going through this, I have one big question: why do we need `Collection` at all? It seems like a very restrictive array-like, which artificially limits items to be of the same type. I don't quite see where we use the class either... is this a leftover of older times?

**Comment by tomvanmele** [comment]
it is a left-over. it could be useful for applying transformations etc to large collections of objects, and potentially in the future in relation to the viewer (it has a collection object for large numbers of small objects to make sure we don't create many small buffers).

but i am happy to remove it

**Pull request closed without merging**